### PR TITLE
Prepare the topi tests for AArch64 CI.

### PR DESCRIPTION
**Pull request by u99127**
This pull request cleans up the TOPI testuite for use on the AArch64
CI target by doing the following:

- Introducing a script to run the tests on AArch64 with a suitable
  invocation of the llvm target string by setting the TVM_TEST_TARGETS
  environment variable.

- Cleaning up the use of hard coded targets and moving the testsuite
  to testing more sensibly with the use of tvm.testing.enabled_targets.

- Cleanup the use of tvm.target.create.

Putting this up for a test run on ci_gpu and ci_cpu to see the effects
of moving TOPI test runs to AArch64 CPU before firing up the Jenkins
changes.

The motivation was from #8361 to pipeclean and add this support.

### CODE REVIEW

File: tests/python/topi/python/test_topi_sparse.py [hunk 1]
```diff
@@ -73,12 +68,12 @@ def check_device(device):
...
         assert a.indptr.dtype == A.indptr.dtype
-        f = tvm.build(s, [nr, A.data, A.indices, A.indptr, B, C, D], device, name="csrmv")
+        f = tvm.build(s, [nr, A.data, A.indices, A.indptr, B, C, D], target, name="csrmv")
         f(_nr, a.data, a.indices, a.indptr, b, c, d)
         tvm.testing.assert_allclose(d.numpy(), d_np, rtol=1e-4, atol=1e-4)
 
-    for device in ["llvm"]:
-        check_device(device)
+    for target, dev in tvm.testing.enabled_targets():
+        check_device(target, dev)
```
Comment (jcf94) [inline tests/python/topi/python/test_topi_sparse.py:76]: Seems the schedules for sparse ops are not ready for GPU? I see the CI failed in cases like this.
We can add a TODO notes here.

Comment (u99127) [reply tests/python/topi/python/test_topi_sparse.py:76]: Yes agreed. there's a few of them like that which needs to be fixed. WIP.

File: tests/python/topi/python/test_topi_conv2d_nhwc.py [hunk 2]
```diff
@@ -66,21 +60,25 @@ def get_ref_data():
...
         w = tvm.nd.array(w_np, dev)
         b = tvm.nd.array(np.zeros(get_const_tuple(B.shape), dtype=B.dtype), dev)
-        func = tvm.build(s, [A, W, B], target)
+        func = tvm.build(s, [A, W, B], device)
         func(a, w, b)
         tvm.testing.assert_allclose(b.numpy(), b_np, rtol=1e-5)
 
-    for target, dev in tvm.testing.enabled_targets():
-        check_device(target, dev)
+    for device in ["llvm", "cuda"]:
```
Comment (areusch) [inline tests/python/topi/python/test_topi_conv2d_nhwc.py:80]: should these become @tvm.testing.requires_?

File: tests/python/topi/python/test_topi_conv2d_nhwc_pack_int8.py [hunk 3]
```diff
@@ -51,26 +52,25 @@ def get_ref_data():
...
         b = tvm.nd.array(np.zeros(get_const_tuple(B.shape), dtype=B.dtype), dev)
-        func = tvm.build(s, [A, W, B], device)
+        func = tvm.build(s, [A, W, B], target)
         func(a, w, b)
         tvm.testing.assert_allclose(b.numpy(), b_np, rtol=1e-5)
 
     # for device in ['llvm -mcpu=skylake-avx512']:
-    for device in ["llvm"]:
-        check_device(device)
+    for target, dev in tvm.testing.enabled_targets():
```
Comment (areusch) [inline tests/python/topi/python/test_topi_conv2d_nhwc_pack_int8.py:72]: same question--possible to use tvm.testing.parametrize_targets ?

File: tests/python/topi/python/test_topi_sparse.py [hunk 4]
```diff
@@ -72,12 +67,15 @@ def check_device(device):
...
         assert a.indices.dtype == A.indices.dtype
         assert a.indptr.dtype == A.indptr.dtype
-        f = tvm.build(s, [nr, A.data, A.indices, A.indptr, B, C, D], device, name="csrmv")
+        f = tvm.build(s, [nr, A.data, A.indices, A.indptr, B, C, D], target, name="csrmv")
         f(_nr, a.data, a.indices, a.indptr, b, c, d)
         tvm.testing.assert_allclose(d.numpy(), d_np, rtol=1e-4, atol=1e-4)
 
-    for device in ["llvm"]:
-        check_device(device)
+    # Unable to use the following idiom and thus restrict to llvm only.
```
Comment (areusch) [inline tests/python/topi/python/test_topi_sparse.py:74]: can you comment why?

File: tests/python/topi/python/test_topi_sparse.py [hunk 5]
```diff
@@ -102,25 +100,23 @@ def get_ref_data():
...
         d = tvm.nd.array(np.zeros((_nr, out_dim), dtype=dtype), dev)
-        f = tvm.build(s, [nr, A.data, A.indices, A.indptr, B, C, D], device, name="csrmm")
+        f = tvm.build(s, [nr, A.data, A.indices, A.indptr, B, C, D], target, name="csrmm")
 
         f(_nr, a.data, a.indices, a.indptr, b, c, d)
         tvm.testing.assert_allclose(d.numpy(), d_np, rtol=1e-2, atol=1e-2)
 
-    for device in ["llvm"]:
-        check_device(device)
+    # This test is not yet ready to use the idiom ,
```
Comment (areusch) [inline tests/python/topi/python/test_topi_sparse.py:115]: could you say what's missing in the comment?

File: tests/python/topi/python/test_topi_sparse.py [hunk 6]
```diff
@@ -594,7 +580,11 @@ def check_device(device):
         )
         tvm.testing.assert_allclose(Y_tvm.numpy(), Y_np.astype("float32"), atol=1e-4, rtol=1e-4)
 
-    check_device("llvm")
+    # Unable to use the idiom below as the test isn't suitable for non llvm targets.
```
Comment (areusch) [inline tests/python/topi/python/test_topi_sparse.py:583]: same question here

**Review by areusch — CHANGES_REQUESTED** [review]
thanks for working on this @u99127 !

**Comment by masahi** [comment]
@u99127 Can you update or close this PR?

**Comment by u99127** [comment]
Abandoned in favour of other changes from @Mousius and others fixing this for AArch64.

**Pull request closed without merging**